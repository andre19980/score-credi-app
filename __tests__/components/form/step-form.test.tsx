import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import StepForm from "@/components/form/step-form";

describe("StepForm component", () => {
  it("renders form", () => {
    const { container } = render(<StepForm />);
    expect(container).toMatchSnapshot();
  });
});
