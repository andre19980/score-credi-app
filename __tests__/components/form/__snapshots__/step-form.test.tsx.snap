// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepForm component renders form 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center gap-y-16 sm:col-start-2 sm:col-span-2"
  >
    <ol
      class="flex items-center w-full px-4 sm:px-16"
    >
      <li
        class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
      >
        <span
          class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
        >
          <svg
            aria-hidden="true"
            class="size-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 6a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0ZM4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.933 17.933 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </li>
      <li
        class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
      >
        <span
          class="bg-gray-100 dark:bg-gray-700 text-gray-500 dark:text-gray-100 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
        >
          <svg
            aria-hidden="true"
            class="size-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6.75V15m6-6v8.25m.503 3.498 4.875-2.437c.381-.19.622-.58.622-1.006V4.82c0-.836-.88-1.38-1.628-1.006l-3.869 1.934c-.317.159-.69.159-1.006 0L9.503 3.252a1.125 1.125 0 0 0-1.006 0L3.622 5.689C3.24 5.88 3 6.27 3 6.695V19.18c0 .836.88 1.38 1.628 1.006l3.869-1.934c.317-.159.69-.159 1.006 0l4.994 2.497c.317.158.69.158 1.006 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </li>
      <li
        class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
      >
        <span
          class="bg-gray-100 dark:bg-gray-700 text-gray-500 dark:text-gray-100 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
        >
          <svg
            aria-hidden="true"
            class="size-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6v12m-3-2.818.879.659c1.171.879 3.07.879 4.242 0 1.172-.879 1.172-2.303 0-3.182C13.536 12.219 12.768 12 12 12c-.725 0-1.45-.22-2.003-.659-1.106-.879-1.106-2.303 0-3.182s2.9-.879 4.006 0l.415.33M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </li>
    </ol>
    <div
      class="bg-gray-50 w-full p-6 rounded"
    >
      <form>
        <div
          style="opacity: 0; will-change: opacity;"
        >
          <h2
            class="text-base font-semibold leading-7 text-gray-900 mb-3"
          >
            Dados Pessoais
          </h2>
          <div
            class="grid grid-cols-1 gap-x-6 gap-y-4 sm:grid-cols-6 mt-4"
          >
            <div
              class="sm:col-span-4"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="name"
              >
                Nome
              </label>
              <div
                class="mt-2"
              >
                <div
                  class="flex bg-white rounded-md shadow-sm ring-1 ring-inset ring-gray-300 focus-within:ring-2 focus-within:ring-inset focus-within:ring-indigo-600"
                >
                  <input
                    class="block flex-1 border-0 bg-transparent py-1.5 pl-1 text-gray-900 placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
                    id="name"
                    name="name"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              class="sm:col-span-2"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="age"
              >
                Idade
              </label>
              <div
                class="mt-2"
              >
                <div
                  class="flex w-full bg-white rounded-md shadow-sm ring-1 ring-inset ring-gray-300 focus-within:ring-2 focus-within:ring-inset focus-within:ring-indigo-600"
                >
                  <input
                    class="block flex-1 w-full border-0 bg-transparent py-1.5 pl-1 text-gray-900 placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
                    id="age"
                    name="age"
                    type="number"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex justify-between pt-4 mt-8 border-t"
        >
          <button
            class="text-gray-600 bg-gray-100 hover:text-gray-600 py-2.5 px-5 text-sm font-medium focus:outline-none rounded-lg border border-gray-200 hover:bg-gray-100 focus:z-10 focus:ring-4 focus:ring-gray-100 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700"
            disabled=""
            type="button"
          >
            Voltar
          </button>
          <button
            class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
            type="button"
          >
            Pr√≥ximo
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
